### PR TITLE
Fstab gatherer

### PR DESCRIPTION
**Pull request by CDimonaco**
This gatherer reads and parse the fstab file.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,6 +31,7 @@ require github.com/tredoe/osutil v1.0.6
 
 require (
 	github.com/chigopher/pathlib v1.0.0 // indirect
+	github.com/d-tux/go-fstab v0.0.0-20141204152952-eb4090f26517 // indirect
```
Comment (arbulu89) [inline go.mod:34]: Just golang  curiosity, why is this `indirect`?

File: internal/factsengine/gatherers/fstab.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	FstabFileDecodingError = entities.FactGatheringError{
+		Type:    "fstab-decoding-error",
+		Message: "error decoding fstab file",
+	}
+)
+
+type FstabEntry struct {
+	Device     string   `json:"device"`
+	MountPoint string   `json:"mount_point"`
+	FS         string   `json:"fs"`
```
Comment (arbulu89) [inline internal/factsengine/gatherers/fstab.go:33]: I would rename this to `FileSystemType` or `Type` (`file_system_type/type`).
It makes it more obvious

File: internal/factsengine/gatherers/fstab.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+func NewFstabGatherer(filePath string) *FstabGatherer {
+	return &FstabGatherer{fstabFilePath: filePath}
+}
+
+func NewDefaultFstabGatherer() *FstabGatherer {
+	return &FstabGatherer{fstabFilePath: FstabFilePath}
+}
+
+func (g *FstabGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
```
Comment (arbulu89) [inline internal/factsengine/gatherers/fstab.go:51]: Maybe update this `g` by `f`

File: internal/factsengine/gatherers/fstab.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		return nil, err
+	}
+
+	var unmarshalled []interface{}
+	err = json.Unmarshal(marshalled, &unmarshalled)
+	if err != nil {
+		return nil, err
+	}
+
+	return entities.NewFactValue(unmarshalled, entities.WithStringConversion())
```
Comment (arbulu89) [inline internal/factsengine/gatherers/fstab.go:99]: We don't need the `WithStringConversion`

File: test/fixtures/gatherers/fstab.valid [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# test fstab
+
+/dev/system/root  /                       btrfs  defaults                      0  0
+/dev/system/root  /root                   btrfs  subvol=/@/root                1  1
+/dev/system/root  /home                   btrfs  subvol=/@/home                1  0
+/dev/system/swap  swap                    swap   defaults                      0  0
+/dev/system/root  /.snapshots             btrfs  subvol=/@/.snapshots          0  1
+UUID=DA2F-21CF    /boot/efi               vfat   utf8
```
Comment (arbulu89) [inline test/fixtures/gatherers/fstab.valid:8]: Is this correct? The last entry doesn't have the `Backup` and `Order`?
I see that in the tests they are set to `0` both of them

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Overall great!
You forgot to add the new gatherer in the `gatherers.go` file.

File: test/fixtures/gatherers/fstab.valid (hunk 5 above)
Comment (arbulu89) [reply test/fixtures/gatherers/fstab.valid:8]: Auto-answer: Yes, they are optional, and they default to 0 if they are not present:
https://www.man7.org/linux/man-pages/man5/fstab.5.html

File: go.mod (hunk 1 above)
Comment (CDimonaco) [reply go.mod]: An indirect dependency is a dependency added but not used, I added the dependency via go get, and I was not using before I wrote the code.

After you wrote the code the dependency is used and it's not indirect anymore, but go build and go run does not update the go mod file.

You should call directly go mod tidy to fix this type of inconsistency in the go mod, I've done that, thanks for spotting! :D

File: internal/factsengine/gatherers/fstab.go (hunk 2 above)
Comment (CDimonaco) [reply internal/factsengine/gatherers/fstab.go]: done

**Review by arbulu89 — APPROVED** [review]
Greeeat!

File: internal/factsengine/gatherers/fstab.go (hunk 3 above)
Comment (CDimonaco) [reply internal/factsengine/gatherers/fstab.go]: g was "gatherer", but I renamed to f, no hard feelings on that :D

File: internal/factsengine/gatherers/fstab.go (hunk 4 above)
Comment (CDimonaco) [reply internal/factsengine/gatherers/fstab.go]: done, damn c&p!

**Pull request merged**